pkg/components/gtime/gtime.go

```diff
@@ -32,5 +32,47 @@ func ParseInterval(interval string) (time.Duration, error) {
 		return now.Sub(now.AddDate(-num, 0, 0)), nil
 	}
 
-	return 0, fmt.Errorf("ParseInterval: invalid duration %q", interval)
+	return 0, fmt.Errorf("invalid interval %q", inp)
+}
+
+// ParseDuration parses a duration with support for all units that Grafana uses.
+func ParseDuration(inp string) (time.Duration, error) {
+	num, period, err := parse(inp)
+	if err != nil {
+		return 0, err
+	}
+	if period == "" {
+		return time.Duration(num), nil
+	}
+
+	// The average number of days in a year, using the Julian calendar
+	const daysInAYear = 365.25
+
+	switch period {
+	case "d":
+		return time.Duration(num*24) * time.Hour, nil
+	case "w":
+		return time.Duration(num*24*7) * time.Hour, nil
+	case "M":
+		return time.Duration(int64(float64(num*24)*(daysInAYear/12))) * time.Hour, nil
+	case "y":
+		return time.Duration(int64(float64(num*24)*daysInAYear)) * time.Hour, nil
```

Comment:
```suggestion
		return time.Duration(float64(num*24)*daysInAYear/12) * time.Hour, nil
	case "y":
		return time.Duration(float64(num*24)*daysInAYear) * time.Hour, nil
```

- `float64` can be cast to `time.Duration` directly.
- Order of operation for `*` and `/` does not matter, removed parens.

https://play.golang.org/p/og8MyCyzqSB